src/jemalloc.c

```diff
@@ -3260,6 +3245,7 @@ je_rallocx(void *ptr, size_t size, int flags) {
 		if (unlikely(usize == 0 || usize > SC_LARGE_MAXCLASS)) {
 			goto label_oom;
 		}
+		thread_event(tsd, usize);
```

Comment:
This looks tricky. The usize might be changed down the irallocx_prof path (note that &usize is passed down). I don't see a clean way to deal with it. Opinions?

Reply:
Yes it is tricky, and even more in the `xallocx()` case. I think we might have to either live with the inaccuracies, or implement some kind of `thread_event_rollback()` mechanism to handle these cases.